test/unit/mdc-select/mdc-select-enhanced.test.js

```diff
@@ -0,0 +1,654 @@
+  domEvents.emit(selectedText, 'focus');
+  td.verify(component.foundation_.handleFocus(), {times: 1});
+});
+
+test('blur event triggers foundation.handleBlur()', () => {
+  const {component, selectedText} = setupTest();
+  component.foundation_.handleBlur = td.func();
+  domEvents.emit(selectedText, 'blur');
+  td.verify(component.foundation_.handleBlur(), {times: 1});
+});
+
+test('#destroy removes the change handler', () => {
+  const {component, selectedText} = setupTest();
+  component.foundation_.handleChange = td.func();
+  component.destroy();
+  domEvents.emit(selectedText, 'change');
+  td.verify(component.foundation_.handleChange(), {times: 0});
+});
+
+test('#destroy removes the focus handler', () => {
+  const {component, selectedText} = setupTest();
+  component.foundation_.handleFocus = td.func();
+  component.destroy();
+  domEvents.emit(selectedText, 'focus');
+  td.verify(component.foundation_.handleFocus(), {times: 0});
+});
+
+test('#destroy removes the blur handler', () => {
+  const {component, selectedText} = setupTest();
+  component.foundation_.handleBlur = td.func();
+  component.destroy();
+  domEvents.emit(selectedText, 'blur');
+  td.verify(component.foundation_.handleBlur(), {times: 0});
+});
+
+test('mousedown on the select sets the line ripple origin', () => {
+  const {bottomLine, fixture} = setupTest();
+  const event = document.createEvent('MouseEvent');
+  const clientX = 200;
+  const clientY = 200;
+  // IE11 mousedown event.
+  event.initMouseEvent('mousedown', true, true, window, 0, 0, 0, clientX, clientY, false, false, false, false, 0, null);
+  fixture.querySelector('.mdc-select__selected-text').dispatchEvent(event);
+
+  td.verify(bottomLine.setRippleCenter(200), {times: 1});
+});
+
+test('mousedown on the select does nothing if the it does not have a lineRipple', () => {
+  const hasOutline = true;
+  const {bottomLine, fixture} = setupTest(hasOutline);
+  const event = document.createEvent('MouseEvent');
+  const clientX = 200;
+  const clientY = 200;
+  // IE11 mousedown event.
+  event.initMouseEvent('mousedown', true, true, window, 0, 0, 0, clientX, clientY, false, false, false, false, 0, null);
+  fixture.querySelector('.mdc-select__selected-text').dispatchEvent(event);
+
+  td.verify(bottomLine.setRippleCenter(200), {times: 0});
+});
+
+test('#destroy removes the mousedown listener', () => {
+  const {bottomLine, component, fixture} = setupTest();
+  const event = document.createEvent('MouseEvent');
+  const clientX = 200;
+  const clientY = 200;
+
+  component.destroy();
+  // IE11 mousedown event.
+  event.initMouseEvent('mousedown', true, true, window, 0, 0, 0, clientX, clientY, false, false, false, false, 0, null);
+  fixture.querySelector('.mdc-select__selected-text').dispatchEvent(event);
+
+  td.verify(bottomLine.setRippleCenter(200), {times: 0});
+});
+
+test('menu surface opened event causes the first element (if not element is selected) to be focused', () => {
```

Comment:
You have tests for the opened event but we need one for the menu selected event (its handler is currently not covered).